spec/fluent.ebnf

```diff
@@ -21,7 +21,9 @@ _                    ::= inline-space
 NL                   ::= line-break
 __                   ::= break-indent
 
-identifier           ::= [a-zA-Z] [a-zA-Z0-9_-]*
+public-identifier    ::= [a-zA-Z] [a-zA-Z0-9_-]*
+private-identifier   ::= '-'+ public-identifier
+identifier           ::= public-identifier | private-identifier
 external             ::= '$' identifier
```

Comment:
I think you need to change `external` to only accept `public-identifier`. Or, maybe I'd even change `public-identifier` to `identifier`, and then create `message-identifier ::= identifier | private-identifier`.

Reply:
Good call, I like this approach better. Thanks.